src/main/lrsql/init/oidc.clj

```diff
@@ -167,6 +177,30 @@
                aud
                (first aud)))))
 
+(def default-authority-path
+  "lrsql/config/oidc_authority.json.template")
+
+(def sample-authority-fn-input
+  {:scope "openid all"
+   :iss   "http://example.com/realm"
+   :sub   "1234"
+   :lrsql/resolved-client-id "someapp"})
+
+(defn- valid-authority-fn?
+  [authority-fn]
+  (s/valid? ::xs/tlo-group ; 3-legged OAuth group
+            (authority-fn sample-authority-fn-input)))
+
+(defn- validate-authority-fn
+  ([authority-fn]
+   (validate-authority-fn authority-fn default-authority-path))
+  ([authority-fn template-path]
+   (if (valid-authority-fn? authority-fn)
+     authority-fn
+     (throw (ex-info "Authority template does not produce a valid xAPI Group for 3-legged OAuth"
+                     {:type          ::invalid-json
+                      :template-path template-path})))))
```

Comment:
I'm not convinced that we should validate the OIDC template beyond ensuring that it is an xAPI agent OR Group, at least not with a hard failure. Maybe a warning?

Reply:
We definitely have a general `:statement/authority` spec so we could do that.

Reply:
https://github.com/yetanalytics/lrsql/pull/248/commits/2161136453aed160d60e1bc5c00702a37dd4ec74